### PR TITLE
Fixing pattern matching for binding patterns

### PR DESCRIPTION
**Pull request by jlahoda**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -485,111 +485,58 @@ matrix:
...
 
-        - name: Test refactoring.java without nb-javac on Java 15
+        - name: Test java.hints batch1 without nb-javac on Java 17
```
Comment (jtulach) [inline .travis.yml:515]: Dropping testing on JDK15 in favor of JDK17 is great!

File: .travis.yml [hunk 2]
```diff
@@ -485,111 +485,58 @@ matrix:
...
...-UNNAMED  --add-opens=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED --add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED --add-opens=java.desktop/javax.swing.text=ALL-UNNAMED --add-opens=jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED --add-exports=jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED --add-opens=jdk.jdeps/com.sun.tools.javap=ALL-UNNAMED --add-exports=jdk.jdeps/com.sun.tools.javap=ALL-UNNAMED --add-opens=java.base/java.security=ALL-UNNAMED --add-exports=java.base/java.security=ALL-UNNAMED --add-opens=java.base/sun.reflect.annotation=ALL-UNNAMED --add-exports=java.base/sun.reflect.annotation=ALL-UNNAMED" test
```
Comment (jtulach) [inline .travis.yml:526]: That's a very long command line. Consider adding a shell script to `nbbuild/travis/*.sh` and just invoking it.

**Review by jtulach — APPROVED** [review]

**Comment by matthiasblaesing** [comment]
Works for me - thank you.

File: .travis.yml [hunk 3]
```diff
@@ -485,111 +485,58 @@ matrix:
...
 
-        - name: Test Java modules without nb-javac on Java 14
+        - name: Test Java modules without nb-javac on Java 17
           jdk: openjdk8
           env:
             - OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -quiet -Dcluster.config=java -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -Dtest-unit-sys-prop.ignore.random.failures=true"
           before_script:
             - wget https://raw.githubusercontent.com/sormuras/bach/master/install-jdk.sh
-            - export TEST_JDK=`bash install-jdk.sh --feature 14 --license GPL --emit-java-home --silent | tail -1`
-            - export OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json $OPTS -Dtest.nbjdk.home=$TEST_JDK -Dtest.use.jdk.javac=true"
```
Comment (jtulach) [inline .travis.yml:495]: Can this change be merged anytime soon?

**Comment by matthiasblaesing** [comment]
@jlahoda could you please have another look at this? This needs to be rebased after the merge of the javac work from @jtulach  and then I suggest to get this merged after.

File: .travis.yml (hunk 2 above)
Comment (mbien) [reply .travis.yml:526]: having the module list in one place might be useful in general. I think the same list has to be updated in netbeans.conf periodically. (not wanting to imply it has to be done in this PR)

Comment (mbien) [reply .travis.yml:526]: @jlahoda we do have this list now (https://github.com/apache/netbeans/tree/master/nbbuild/jms-config), however feel free to keep the flags in the yml for now if it fixes the tests, we can clean this up later.

**Pull request closed without merging**